### PR TITLE
Add vulnerability detector capability rework

### PR DESCRIPTION
**Pull request by javimed**
## Description
This PR updates `/user-manual/capabilities/vulnerability-detection/` with the rework of the whole section.

## Checks
- [x] Compiles without warnings.
- [x] Uses present tense, active voice, and semi-formal registry.
- [x] Uses short, simple sentences.
- [x] Uses **bold** for user interface elements, _italics_ for key terms or emphasis, and `code` font for Bash commands, file names, REST paths, and code.
- [x] Uses three spaces indentation.
- [x] Adds or updates meta descriptions accordingly.
- [x] Updates the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

### CODE REVIEW

File: source/_static/js/redirects.js [hunk 1]
```diff
@@ -71,7 +71,16 @@ removedUrls['x.y'] = [
 
 /* *** RELEASE 4.3 ****/
 
-/* Pages added in 4.3 */
+/* Pages changed within the same release */
+
+redirectSameRelease['4.3'] = {
+   '/user-manual/capabilities/vulnerability-detection/compatibility-matrix.html':
+     '/user-manual/capabilities/vulnerability-detection/how-it-works.html',
```
Comment (javimed) [inline source/_static/js/redirects.js:78]:
```suggestion
     '/user-manual/capabilities/vulnerability-detection/how-it-works.html#compatibility-matrix',
```

File: source/user-manual/capabilities/vulnerability-detection/allow-os.rst [hunk 2]
```diff
@@ -1,92 +1,98 @@
...
-
-  It is necessary to include the operating system using the ``allow`` attribute followed by the corresponding feed, following the syntax: "OS name-major version." To include more than one system, just add a list of systems separated by commas.
+   You need the name and major operating system version to configure the Vulnerability Detector.
 
-  For example, if we want to force the compatibility of Ubuntu 16 (Xenial) feed with PopOS 16, and Ubuntu 18 (Bionic) with Linux Mint 19 and Pop OS 18, we will use:
+Configuring Vulnerability Detector to include unsupported systems
+-----------------------------------------------------------------
 
-    .. code-block:: xml
+Use this format: ``OS_name-OS_major`` with the ``allow`` attribute to include the operating system. Add a list of systems separated by commas to include more than one operating system.
```
Comment (Arcapas) [inline source/user-manual/capabilities/vulnerability-detection/allow-os.rst:76]:
```suggestion
Use the format ``OS_name-OS_major`` with the ``allow`` attribute to include the operating system. Add a list of systems separated by commas to include more than one operating system.
```

File: source/user-manual/capabilities/vulnerability-detection/configuring-scans.rst [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+You can see the inventory of all the vulnerable packages installed for a particular agent by clicking on the **Vulnerabilities** module on the WUI and selecting the agent.
+
+.. thumbnail:: /images/manual/vuln-detector/vuln-inventory01.png
+   :title: Vulnerable packages inventory
+   :align: center
+   :width: 80%
+
+.. thumbnail:: /images/manual/vuln-detector/vuln-inventory02.png
+   :title:
```
Comment (Arcapas) [inline source/user-manual/capabilities/vulnerability-detection/configuring-scans.rst:228]: This image doesn't have a title.

File: source/user-manual/capabilities/vulnerability-detection/configuring-scans.rst [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+.. thumbnail:: /images/manual/vuln-detector/vuln-inventory02.png
+   :title: 
+   :align: center
+   :width: 80%
+
+Here you can see what certain fields of the alert look like on the dashboard:
+
+.. thumbnail:: /images/manual/vuln-detector/vuln-alert-fields.png
+   :title:
```
Comment (Arcapas) [inline source/user-manual/capabilities/vulnerability-detection/configuring-scans.rst:235]: This image doesn't have a title.

File: source/user-manual/capabilities/vulnerability-detection/scan-types.rst [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+-  **Full scan**: The vulnerability detector scans every installed package and operating system in this scan type. It runs only when the configured ``min_full_scan_interval`` expires and when the CVEs database contains new information. As a result, Wazuh generates alerts when there is any update/change in the vulnerability inventory.
+-  **Partial scan**: The vulnerability detector only scans new packages. As a result, Wazuh generates alerts when there is any update/change in the CVE inventory.
+
+A few considerations arise from this behavior:
+
+-  The :ref:`min_full_scan_interval <vuln_det_min_full_scan_interval>` setting protects the manager performance by not running Full scans too often, especially when the manager receives many updates to the vulnerabilities feeds.
+-  Every vulnerability in the agent vulnerabilities inventory is in three different states:
+
+   -  *VALID*: Indicates that the vulnerability is still present in the system.
+   -  *PENDING*: A Full scan is in progress, and the vulnerability needs to be confirmed.
```
Comment (Arcapas) [inline source/user-manual/capabilities/vulnerability-detection/scan-types.rst:21]:
```suggestion
   -  **PENDING**: A Full scan is in progress, and the vulnerability needs to be confirmed.
```

File: source/user-manual/capabilities/vulnerability-detection/scan-types.rst [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+   -  *OBSOLETE*: Indicates that the vulnerability is no longer present in the system. The vulnerability detector generates removal alerts when any vulnerability enters this state.
```
Comment (Arcapas) [inline source/user-manual/capabilities/vulnerability-detection/scan-types.rst:22]:
```suggestion
   -  **OBSOLETE**: Indicates that the vulnerability is no longer present in the system. The vulnerability detector generates removal alerts when any vulnerability enters this state.
```

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 7]
```diff
@@ -3,236 +3,220 @@
...
-
-Every single program in the Windows agents must be covered in this dictionary to be scanned for vulnerabilities.
-
-- `CPE Helper Schema`_
-- `Dictionary schema`_
-- `Translation types`_
-- `How to translate a Windows program to CPE format`_
-- `Combine several programs in a dictionary entry`_
-- `Products whose version does not change between updates`_
+Users must add the applications installed on Windows endpoints to this dictionary to enable the Vulnerability Detector module scan the applications.
```
Comment (Arcapas) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:11]:
```suggestion
Users must add the applications installed on Windows endpoints to this dictionary to enable the Vulnerability Detector module to scan the applications.
```

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 8]
```diff
@@ -3,236 +3,220 @@
...
 -----------------
 
-Each dictionary entry has the following fields:
+Each dictionary entry has the following fields to describe the application that the user wants to add:
 
-- **target**: Operating system affected by this entry. At the moment, only Windows is available.
+-  **target**: Operating system of the application you want to add. Currently, Wazuh only uses the CPE helper dictionary for applications installed on Windows endpoints.
 
-- **source**: Parameters of the program to be translated. Allowed sections:
+-  **source**: Parameters of the application you want to translate. The allowed sections are:
```
Comment (Arcapas) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:51]:
```suggestion
-  **source**: Parameters of the application you want to translate. The allowed sections are the following:
```

File: source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst [hunk 9]
```diff
@@ -3,236 +3,220 @@
...
-  - **target_hw (optional)**: Defines a pattern to extract the ``target_hw`` field. Example *^Python [0-9]+\\.*[0-9]*\\.*[0-9]* \\(([6432]{2}-bit)\\)*.
+   -  **vendor**: Array of one or more vendor fields that match the dictionary entry. You can use a regular expression. For example, *^Oracle*.
+   -  **product**: Array of one or more product fields that match the dictionary entry. You can use a regular expression. For example, *^Oracle.*VirtualBox [0-9]*.
+   -  **version (optional)**: Array of one or more version fields that match the dictionary entry. You can also define a pattern to match the software version using regular expressions. For example *^Microsoft Office.*([0-9]{4})*.
+   -  **target_hw (optional)**: This field defines the target hardware of the application. It can also be a regular expression. For example, *^Python [0-9]+\.*[0-9]\.*[0-9]* \(([6432]{2}-bit)\)**.
 
-- **translation**: Section to specify the CPE translation. Allowed sections:
+-  **translation**: Section to specify the CPE translation. The allowed sections are:
```
Comment (Arcapas) [inline source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst:58]:
```suggestion
-  **translation**: Section to specify the CPE translation. The allowed sections are the following::
```

**Review by Arcapas — COMMENTED** [review]
I made some suggestions.

File: source/user-manual/capabilities/vulnerability-detection/configuring-scans.rst (hunk 3 above)
Comment (javimed) [inline source/user-manual/capabilities/vulnerability-detection/configuring-scans.rst:228]:
```suggestion
   :title: Vulnerability details in Vulnerable packages inventory
```

File: source/user-manual/capabilities/vulnerability-detection/configuring-scans.rst (hunk 4 above)
Comment (javimed) [inline source/user-manual/capabilities/vulnerability-detection/configuring-scans.rst:235]:
```suggestion
   :title: Vulnerability alert fields
```

**Review by Arcapas — APPROVED** [review]
LGTM

**Pull request merged**